js/store.js

```diff
@@ -54,12 +56,24 @@ export class TodoStore extends EventTarget {
 			}, async (update, state) => {
 				updateSubject({'@id': 'todos', '@list': this.todos}, update);
 				await loadTodoReferences(this.todos, state);
-				this.dispatchEvent(new CustomEvent("save"));
+				this.dispatchEvent(new CustomEvent("save", {
+					// We include the actual update made as a detail of the save
+					// event so that the app can adjust the caret and selection
+					detail: {update, isEcho: this._updating}
+				}));
 			});
 		}).catch(this._handleError);
 	}
 	_save(write) {
-		this.meld.write(write).catch(this._handleError);
+		this.meld.write(async (state) => {
+			// This allows us to tell the app whether an update is an echo
+			this._updating = true;
+			try {
+				await state.write(write);
+			} finally {
+				this._updating = false;
+			}
+		}).catch(this._handleError);
```

Comment:
It's slightly awkward to have to pass the update itself (and the echo flag) through to the App so that it can correctly adjust the caret position.